RFCs/FS-1004-result-type.md

```diff
@@ -48,3 +48,42 @@ The standard range of testing considerations for F# library types apply.
 - declare as struct
 - compiled name
 - require additional functions?
+
+### Proposal for additional functions
+
+Minimal:
+
+```fsharp
+bind : ('TSuccess -> Result<'T, 'TError>) -> Result<'TSuccess, 'TError> -> Result<'T, 'TError>
```

Comment:
@vasily-kirichenko The F# guidelines clearly recommend ``'t`` named type variables, and it's standard throughout FSharp.Core.  

But I would prefer ``Result<'T,'TError`>, I don't think ``'TSuccess`` adds anything over ``'T``